src/main/java/com/google/devtools/build/lib/starlarkbuildapi/repository/StarlarkRepositoryContextApi.java

```diff
@@ -329,8 +329,9 @@ void patch(Object patchFile, StarlarkInt strip, StarlarkThread thread)
       name = "download",
       doc =
           "Downloads a file to the output path for the provided url and returns a struct"
-              + " containing a hash of the file with the fields <code>sha256</code> and"
-              + " <code>integrity</code>.",
+              + " containing <code>success</code>, a flag which is <code>true</code> if the"
+              + " download completed successfully, and if successful, a hash of the file"
+              + " with the fields <code>sha256</code> and <code>integrity</code>.",
```

Comment:
"with the fields ..."  reads like the hash is a sub-struct.   {success: t/f,  "hash?": {sha256, integrity}}

Is it actually a single struct with up to 3 fields, where sha256 and integrity are not included if success is false?
I think so, but I do not know the answer offhand.

Reply:
I also do not know the answer offhand.  I will try to research it, but I'm not a contributor to this part of the code so it will take some time for me.